lib/validators/one_favorite_per_owner_validator.rb

```diff
@@ -0,0 +1,16 @@
+module Validators
+  # Validator for Collection model
+  # Validate that only one favorite collection is created per owner for each project
```

Comment:
```suggestion
  # Validate that only one favorite collection is created per owner for each project
  # This validation technique may allow a duplicate favorite record be inserted into the database
  # due to the db the read isolation level of this transaction vs concurrent transactions
  # and the fact our validations exist in ruby land and not strictly enforced at the database layer
  # it's low probablity but worth noting here if folks wonder how it can happen
```

Reply:
>   This validation technique may allow a duplicate favorite record be inserted into the database due to the db the read isolation level of this transaction vs concurrent transactions

Just to confirm that I'm understanding this correctly: 
the hypothetical duplicate favorite scenario you're describing here would occur if two requests to create a new favorite collection come in at the same time? Can you explain further about the read isolation level?